test/extensions/filters/network/thrift_proxy/conn_manager_test.cc

```diff
@@ -1925,6 +1936,127 @@ stat_prefix: test
   EXPECT_EQ(nullptr, route);
 }
 
+TEST_F(ThriftConnectionManagerTest, UnknownCluster) {
+  const std::string yaml = R"EOF(
+transport: FRAMED
+protocol: BINARY
+stat_prefix: test
+route_config:
+  name: "routes"
+  routes:
+    - match:
+        method_name: name
+      route:
+        cluster: cluster
+)EOF";
+
+  EXPECT_THROW_WITH_REGEX(initializeFilter(yaml), EnvoyException, "unknown thrift cluster");
+}
+
+TEST_F(ThriftConnectionManagerTest, UnknownCluster2) {
+  const std::string yaml = R"EOF(
+transport: FRAMED
+protocol: BINARY
+stat_prefix: test
+route_config:
+  name: "routes"
+  routes:
+    - match:
+        method_name: name
+      route:
+        cluster: cluster
+    - match:
+        method_name: name2
+      route:
+        cluster: unknown_cluster
+)EOF";
+
+  EXPECT_THROW_WITH_REGEX(initializeFilter(yaml, {"cluster"}), EnvoyException,
+                          "unknown thrift cluster");
+}
+
+TEST_F(ThriftConnectionManagerTest, UnknownCluster3) {
+  const std::string yaml = R"EOF(
+transport: FRAMED
+protocol: BINARY
+stat_prefix: test
+route_config:
+  name: "routes"
+  routes:
+    - match:
+        method_name: name
+      route:
+        cluster: unknown_cluster
+    - match:
+        method_name: name2
+      route:
+        cluster: cluster
+)EOF";
+
+  EXPECT_THROW_WITH_REGEX(initializeFilter(yaml, {"cluster"}), EnvoyException,
+                          "unknown thrift cluster");
+}
+
+TEST_F(ThriftConnectionManagerTest, UnknownWeightedCluster) {
+  envoy::extensions::filters::network::thrift_proxy::v3::ThriftProxy config;
+  {
+    auto* route_config = config.mutable_route_config();
+    route_config->set_name("config");
+    auto* route = route_config->add_routes();
+    route->mutable_match()->set_method_name("foo");
+    auto* action = route->mutable_route();
+    auto* cluster1 = action->mutable_weighted_clusters()->add_clusters();
+    cluster1->set_name("thrift");
+    cluster1->mutable_weight()->set_value(50);
+    auto* cluster2 = action->mutable_weighted_clusters()->add_clusters();
+    cluster2->set_name("unknown_cluster");
+    cluster2->mutable_weight()->set_value(50);
+  }
+  // Intentionally miss "unknown_cluster" cluster.
+  EXPECT_THROW_WITH_REGEX(initializeFilter(config, {"thrift"}), EnvoyException,
+                          "unknown thrift weighted cluster");
+}
+
+TEST_F(ThriftConnectionManagerTest, UnknownWeightedCluster2) {
+  envoy::extensions::filters::network::thrift_proxy::v3::ThriftProxy config;
```

Comment:
Do we really need this test case?
If yes, then the config initialization is quite copy-paste. It could be extracted to a function like 
`initRouteConfig(..., const std::vector<std::string>& cluster_names, std::vector<int>& weights>)` or something.

Reply:
Again it's just random thought, the weight shouldn't matter. I'll merge these two cases.